numpy/lib/function_base.py

```diff
@@ -1943,11 +1943,19 @@ def _calculate_shapes(broadcast_shape, dim_sizes, list_of_core_dims):
             for core_dims in list_of_core_dims]
 
 
-def _create_arrays(broadcast_shape, dim_sizes, list_of_core_dims, dtypes):
+def _create_arrays(broadcast_shape, dim_sizes, list_of_core_dims, dtypes,
+                   results=None):
     """Helper for creating output arrays in vectorize."""
     shapes = _calculate_shapes(broadcast_shape, dim_sizes, list_of_core_dims)
-    arrays = tuple(np.empty(shape, dtype=dtype)
-                   for shape, dtype in zip(shapes, dtypes))
+    if dtypes is None:
+        dtypes = [None] * len(shapes)
```

Comment:
This branch does not make sense if `results is None`, maybe just move it into that branch?

Reply:
I agree that in practice it is not currently possible to get here, since it either gets called with `results` or with explicit `dtypes`, but in the context of this helper function it seems clearer to just allow `dtypes=None` for both cases (and I need a list to be able to iterate over them).

If you think it better, though, since the function is now effectively different for the two paths insize `vectorize`, I could just inline it there.

Reply:
Doesn't matter, I just noticed it.  What made me pause was just that in that path `dtype=None` effectively gives `dtype=np.dtype(None)` which is `float64`.